bin/pytorchMPSevolve.py

```diff
@@ -0,0 +1,144 @@
+import numpy as np
+import torch
+from torch.autograd import Variable
+
+def init_spinup_MPS(N, D=1):
+    """MPS: all spins up
+    Args
+    ----
+    D : physical dimension, product state for D=1
+    """
+    Ms = np.zeros([N, D, 2, D], dtype=np.csingle)
+    Ms[:, 0, 0, 0] = 1. # spinup
+    Ms = torch.from_numpy(Ms).requires_grad_(True)
+    return MPS(Ms)
+
+def init_random_MPS(N, D=1):
+    """MPS: all spins up
+    Args
+    ----
+    D : physical dimension, product state for D=1
+    """
+    Ms = torch.rand([N, D, 2, D], dtype=torch.cfloat, requires_grad=True)
+    return MPS(Ms)
+
+class MPS:
+    """Class for handling matrix product states assuming a Vidal form"""
+    def __init__(self, Ms):
+        self.Ms = Ms
+        self.N = len(Ms)
+
+    def get_theta1(self, i):
+        """Effective single-site wave function on site i"""
+        return torch.tensordot(torch.diag(self.Ss[i]), self.Ms[i], [[1], [0]])  
```

Comment:
Is `tensordot` faster than `einsum`? If not (very likely), I would suggest using einsum everywhere. Because it is much more readable.

Reply:
Aren't `einsum` and `tensordot` calling the same dot products under the hood? But I agree that `einsum` is nice.

Reply:
I think so, I am thinking about readability. Here, you can not see input tensor dimension for example.

Reply:
FYI, here you should write `einsum("i,ij->ij", self.Ss[i], self.Ms[i])`